src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -827,7 +839,10 @@ static int enumerate_partitions(dev_t devnum) {
                         r = k;
         }
 
-        if (m->partitions[PARTITION_ESP].found) {
+        if (m->partitions[PARTITION_ESP].found &&
+            esp_is_valid(m->partitions + PARTITION_ESP,
```

Comment:
indentation to 4 space is a bit weird, usually we'd indent this deeper (16spaces?), but doesn't matter